### PR TITLE
Deprecate CalibrateRectangularDetectors

### PR DESCRIPTION
**Pull request by wdzhou**
**Description of work.**

Deprecate CalibrateRectangularDetectors and remove its system test.

**To test:**

Visual check
All tests shall be passed.
There is no associated Mantid issue.

It is associated with https://code.ornl.gov/sns-hfir-scse/spectroscopy/spectroscopy/-/issues/117

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v6.3.0/diffraction.rst [hunk 1]
```diff
@@ -12,6 +12,8 @@ Diffraction Changes
 Powder Diffraction
 ------------------
 
+- CalibrateRectangularDetectors is deprecated.
```
Comment (peterfpeterson) [inline docs/source/release/v6.3.0/diffraction.rst:15]: Please link to the algorithm

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Minor change then this looks good to go

**Review by peterfpeterson — APPROVED** [review]

**Review by DavidFair — CHANGES_REQUESTED** [review]
Could we leave the system test in until we remove this?
Whilst it's deprecated I'm concerned about any regressions from unrelated changes, on the off-chance someone is running this

**Comment by wdzhou** [comment]
The purpose of this PR is to deprecate `CalibrateRectangularDetector` because (1) `PDCalibration` has replaced it for years and (2) the algorithm underneath has inconsistent behavior in Ubuntu 20.04.

**Comment by DavidFair** [comment]
> @DavidFair The purpose of this PR is to deprecate `CalibrateRectangularDetector` because (1) `PDCalibration` has replaced it for years and (2) the algorithm underneath has inconsistent behavior in Ubuntu 20.04.

So the hitch for me is deprecate != remove
Whilst it's in Mantid and callable we should have a test to make sure we don't break it (even if nobody is likely calling it) especially if it's already there for free.

If we're 100% sure absolutely nobody uses it in any way shape or form lets skip straight to removal rather than letting it rot

**Comment by wdzhou** [comment]
Task was changed.  Replace by #32500

**Pull request closed without merging**